src/sentry/integrations/bitbucket_server/client.py

```diff
@@ -0,0 +1,139 @@
+        self.verify_ssl = verify_ssl
+
+    def get_request_token(self):
+        """
+        Step 1 of the oauth flow.
+        Get a request token that we can have the user verify.
+        """
+        url = self.request_token_url.format(self.base_url)
+        resp = self.post(url, allow_text=True)
+        return dict(parse_qsl(resp.text))
+
+    def get_authorize_url(self, request_token):
+        """
+        Step 2 of the oauth flow.
+        Get a URL that the user can verify our request token at.
+        """
+        return self.authorize_url.format(self.base_url, request_token["oauth_token"])
+
+    def get_access_token(self, request_token, verifier):
+        """
+        Step 3 of the oauth flow.
+        Use the verifier and request token from step 1 to get an access token.
+        """
+        if not verifier:
+            raise ApiError("Missing OAuth token verifier")
+        auth = OAuth1(
+            client_key=self.consumer_key,
+            resource_owner_key=request_token["oauth_token"],
+            resource_owner_secret=request_token["oauth_token_secret"],
+            verifier=verifier,
+            rsa_key=self.private_key,
+            signature_method=SIGNATURE_RSA,
+            signature_type="auth_header",
+        )
+        url = self.access_token_url.format(self.base_url)
+        resp = self.post(url, auth=auth, allow_text=True)
+        return dict(parse_qsl(resp.text))
+
+    def request(self, *args, **kwargs):
+        """
+        Add OAuth1 RSA signatures.
+        """
+        if "auth" not in kwargs:
+            kwargs["auth"] = OAuth1(
+                client_key=self.consumer_key,
+                rsa_key=self.private_key,
+                signature_method=SIGNATURE_RSA,
+                signature_type="auth_header",
+            )
+        return self._request(*args, **kwargs)
+
+
+class BitbucketServer(ApiClient):
+    """
+    Contains the bitbucket server specifics in order to communicate with bitbucket
+
+    You can find BITBUCKET REST API docs here:
+
+    https://developer.atlassian.com/server/bitbucket/reference/rest-api/
+    """
+
+    integration_name = "bitbucket_server"
+
+    def __init__(self, base_url, credentials, verify_ssl):
+        super(BitbucketServer, self).__init__(verify_ssl)
+
+        self.base_url = base_url
+        self.credentials = credentials
+
+    def get_repos(self):
+        return self.get(
+            "/rest/api/1.0/repos",
+            auth=self.get_auth(),
+            params={"limit": 250, "permission": "REPO_ADMIN"},
+        )
+
+    def search_repositories(self, query_string):
+        return self.get(
+            "/rest/api/1.0/repos",
+            auth=self.get_auth(),
+            params={"limit": 250, "permission": "REPO_ADMIN", "name": query_string},
+        )
+
+    def get_repo(self, project, repo):
+        return self.get(
+            "/rest/api/1.0/projects/%s/repos/%s" % (project, repo), auth=self.get_auth()
```

Comment:
@pierredup Can you create a class that has the path templates as we do with BitBucket Cloud with the `BitbucketAPIPath` class? Would like to be consistent when possible.